src/Core/Components/Overlay/FluentOverlay.razor.cs

```diff
@@ -0,0 +1,128 @@
+// ------------------------------------------------------------------------
+// This file is licensed to you under the MIT License.
+// ------------------------------------------------------------------------
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.AspNetCore.Components;
+using Microsoft.FluentUI.AspNetCore.Components.Utilities;
+using Microsoft.JSInterop;
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+/// <summary>
+/// Represents an overlay component that provides visual layering and interaction blocking in Fluent UI applications.
+/// </summary>
+public partial class FluentOverlay : FluentComponentBase, IFluentComponentElementBase
+{
+    /// <summary />
+    public FluentOverlay(LibraryConfiguration configuration) : base(configuration)
+    {
+        Id = Identifier.NewId();
+    }
+
+    /// <summary />
+    protected string? ClassValue => DefaultClassBuilder
+        .Build();
+
+    /// <summary />
+    protected string? StyleValue => DefaultStyleBuilder
+        .Build();
+
+    /// <inheritdoc cref="IFluentComponentElementBase.Element" />
+    [Parameter]
+    public ElementReference Element { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the component is displayed in full screen mode.
+    /// </summary>
+    [Parameter]
+    public bool FullScreen { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the overlay allows interaction with underlying content.
+    /// </summary>
+    [Parameter]
+    public bool Interactive { get; set; }
+
+    /// <summary>
+    /// Gets or sets the background color of the overlay.
+    /// Default is `var(--colorBackgroundOverlay)`.
+    /// </summary>
+    [Parameter]
+    public string BackgroundColor { get; set; } = "var(--colorBackgroundOverlay)";
+
+    /// <summary>
+    /// Gets or sets the opacity level of the overlay background.
+    /// Default is `0.2` (20%).
+    /// </summary>
+    [Parameter]
+    public double Opacity { get; set; } = 0.4;
```

Comment:
I would consider changing this to 40 and remove the `* 100` logic. It is a breaking change but it makes it much clearer on how to use the component. If you want 40.5% you provide the value 40.5.

Reply:
Done